### PR TITLE
Epic 7 - Maksym Lirko

### PR DESCRIPTION
**Pull request by maksymlirko**

### CODE REVIEW

**Comment by izagor** [comment]
+ 12-19-2023

File: ai_12/maksym_lirko/Epic 7/cr.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            fullColumnSum += matrix[i][j];
+        }
+        
+        sums[j] = fullColumnSum;
+    }
+}
+}
+
+bool writeArrayToFile(int matrix[10][10], const string& filename) {
+    ofstream file(filename); // Відкриваємо файл для запису
```
Comment (fm2lpnu) [inline ai_12/maksym_lirko/Epic 7/cr.cpp:75]: Missing library import to work with files

File: ai_12/maksym_lirko/Epic 7/cr.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    const int pullRequestValue = 941; // константа з номером пул-реквесту
+
+    const string filename = "matrix_values.txt";
+
+    float userInputValue; // Оголошення змінної
+    cout << "Enter a fractional number: ";
+    cin >> userInputValue;
+
+    createMatrix(resultMatrix, pullRequestValue, userInputValue); // виклик створення
+    sortRows(resultMatrix); // виклик сортування
```
Comment (fm2lpnu) [inline ai_12/maksym_lirko/Epic 7/cr.cpp:103]: Task 2 mismatch with requirements.
Overloaded function should execute sort

File: ai_12/maksym_lirko/Epic 7/cr.cpp [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+#include <cmath> 
+
+using namespace std;
+
+int calculateValue(int x, float constant)  // Функція використовується для обчислення за допомогою матем. функц
+{
+    return static_cast<int>(-2.4 * x + 5 * x - constant);
+}
+
+void createMatrix(int matrix[10][10], int pullRequestValue, float userInputValue)
```
Comment (fm2lpnu) [inline ai_12/maksym_lirko/Epic 7/cr.cpp:11]: Task 1 (a,b) is not implemented according to requirements.
One of two options (a,b) is covered.

File: ai_12/maksym_lirko/Epic 7/cr.cpp [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    cout << "Enter a fractional number: ";
+    cin >> userInputValue;
+
+    createMatrix(resultMatrix, pullRequestValue, userInputValue); // виклик створення
+    sortRows(resultMatrix); // виклик сортування
+    printMatrix(resultMatrix); // Виклик функції виведення масиву
+
+    int columnSums[10] = {0}; // Масив для зберігання сум
+
+    calculateAndStoreColumnSums(resultMatrix, columnSums); // Вивід функції яка
```
Comment (fm2lpnu) [inline ai_12/maksym_lirko/Epic 7/cr.cpp:108]: Task 3 is not implemented according to requirements.
Random is not used.

File: ai_12/maksym_lirko/Epic 7/cr.cpp [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    cout << endl;
+
+      bool writeResult = writeArrayToFile(resultMatrix, filename);
+
+    if (writeResult) {
+        cout << "Matrix values were successfully written to the file '" << filename << "'." << endl;
+    } else {
+        cout << "Failed to write matrix values to the file." << endl;
+    }
+
```
Comment (fm2lpnu) [inline ai_12/maksym_lirko/Epic 7/cr.cpp:132]: Task 6 is not implemented

**Review by fm2lpnu — APPROVED** [review]

**Pull request merged**